src/main/java/com/palantir/docker/compose/configuration/RemoteEnvironmentValidator.java

```diff
@@ -23,25 +23,30 @@
 
 import com.google.common.base.Preconditions;
 import com.google.common.base.Strings;
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
 import com.google.common.collect.ImmutableSet;
 import com.google.common.collect.Sets;
 import java.util.Collection;
 import java.util.Map;
 import java.util.Set;
 import java.util.stream.Collectors;
 
-public class RemoteEnvironmentValidator {
+public final class RemoteEnvironmentValidator implements EnvironmentValidator {
 
     private static final Set<String> SECURE_VARIABLES = ImmutableSet.of(DOCKER_TLS_VERIFY, DOCKER_CERT_PATH);
+    private static final Supplier<RemoteEnvironmentValidator> SUPPLIER = Suppliers.memoize(
```

Comment:
Could we get away with just having a regular old static final field here?  How necessary is this memoized supplier?